src/components/nature-of-business/NatureOfBusiness.vue

```diff
@@ -242,6 +242,7 @@ export default class NatureOfBusiness extends Vue {
   @Watch('haveNaics', { immediate: true })
   private onHaveNaicsChanged (val: boolean): void {
     this.state = val ? States.SUMMARY : States.INITIAL
+    this.emitValid(this.haveNaics)
```

Comment:
I'm hesistant about this solution since validity also depends on whether this component is expanded.

Would making line 263 emit an initial value work instead?

Reply:
It does not unfortunately, immediate checks before the filing gets build. I think that this should be safe however since once you chose a NAICS value then you can't set it back to an empty value, so it should be safe.

Reply:
Does this component work correctly in Edit UI? What's different between Create UI and Edit UI?

Reply:
It does however the way the components is used differently. In the Edit UI each component sets validity where as in the Create UI the Registration view sets the validity.

Reply:
I mean, some sort of change in the local, intermediate component NatureOfBusiness.vue?

Reply:
Doing it that way would make NOB the only component to validate like that in registration. Which might not be great

Another solution I could do is use the updated function in the NOB component so it emits the NAICS code after the component has been loaded?

ie ```updated () { this.emitValid(!!this.getRegistration.naics) }``` in the create UI

Reply:
Does the shared NOB component emit an initial status? Is it incorrect? If so, why?

What's the actual problem here -- is the initial validation status incorrect? Not being updated?

Reply:
Sorry for not getting back to you.

So I have been going about it the wrong way. The issue is that the emit https://github.com/bcgov/business-create-ui/blob/7b256588cb28eeb800d7ea807bac02a740a0ddbd/src/views/Registration/RegistrationDefineBusiness.vue#L34
is not running this function https://github.com/bcgov/business-create-ui/blob/7b256588cb28eeb800d7ea807bac02a740a0ddbd/src/views/Registration/RegistrationDefineBusiness.vue#L204

I am trying to figure out why

Reply:
So what is happening is the NOB components only emits once if there is a change to the component and the validity is local to the view and then validity is evaluated upon building the filing. The edit ui gets around this issue by having the validity set and stored in state so when the filing is built the validity of the NOB is loaded as well.

The two ways that I have found to fix this are around this are. 1. Trigger the emit like above. 2. Have a function like mounted or updated trigger the emit when NOB is rendered. This is what StartDate does

Reply:
I had a quick look at this locally. The problem I saw is that the shared NOB component does not emit an initial "valid" event, so RegistrationDefineBusiness.vue doesn't get an initial value (the value stays False).

This seems to work correctly with this change:
![image](https://user-images.githubusercontent.com/10002889/196560687-54df0c03-77e4-41f4-89c7-7b78d8b8f9d8.png)

What do you think?